### PR TITLE
[PyROOT] Add entry about boolean value of C++ instance proxies

### PR DESCRIPTION
**Pull request by etejedor**
Reported by Walter Lampl.

### CODE REVIEW

**Review by stwunsch — CHANGES_REQUESTED** [review]
I would rephrase along the lines of the following proposal:

> When obtaining a boolean value of a C++ instance proxy, new cppyy does an extra check and returns `False` if the C++ pointer is null. Otherwise, the normal Python behaviour is followed (see [here](https://docs.python.org/3.8/reference/datamodel.html#object.__bool__)). For example, the following code shows how the insertion of `__len__` changes the boolean value of an instance proxy:

Would it be possible to create an instance with a zero pointer to show that it's `False`? I cannot think of something nice looking, just ugly tricks.

File: manual/python/index.md [hunk 1]
```diff
@@ -597,3 +597,26 @@ modifier methods on the `std::string` objects.
 <class cppyy.gbl.std.string at 0x4cd41b0>
 <class cppyy.gbl.std.string at 0x4cd41b0>
 ```
+
+- When obtaining the boolean value of a C++ instance proxy, both old and new PyROOT return
+`False` when such proxy points to null. On the other hand, when the proxy points to a C++ object,
```
Comment (stwunsch) [inline manual/python/index.md:602]:
```suggestion
`False` when such proxy points to null. On the other hand, when the proxy points to a valid C++ object,
```

**Review by stwunsch — APPROVED** [review]
Looks good! One small improvement could be done by pointing out that we return `True` for a **valid** C++ object.

File: manual/python/index.md (hunk 1 above)
Comment (etejedor) [reply manual/python/index.md:602]: Indeed, I thought about this, but then I realized that even if the object is not valid (e.g. it has been already deleted without the proxy knowing it), it could still return `True`.
